doc/content/design/sm-supported-image-formats.md

```diff
@@ -22,32 +22,40 @@ available formats.
 # Design Proposal
 
 To expose the available image formats to clients (e.g., XenCenter, XenOrchestra, etc.),
-we propose adding a new field called `supported-image-formats` to the Storage Manager (SM)
-module. This field will be included in the output of the `SM.get_all_records` call.
+we propose adding a new field called `supported_image_formats` to the Storage Manager
+(SM) module. This field will be included in the output of the `SM.get_all_records` call.
 
-The `supported-image-formats` field will be populated by retrieving information
+The `supported_image_formats` field will be populated by retrieving information
 from the SMAPI drivers. Specifically, each driver will update its `DRIVER_INFO`
 dictionary with a new key, `supported_image_formats`, which will contain a list
 of strings representing the supported image formats
 (for example: `["vhd", "raw", "qcow2"]`).
 
-The list designates the driver's preferred VDI format as its first entry. That
-means that when migrating a VDI, the destination storage repository will
-attempt to create a VDI in this preferred format. If the default format cannot
-be used (e.g., due to size limitations), an error will be generated.
+When creating new VDI, you can choose the format by passing the option
+`type=qcow2` to the `sm_config` parameter. If no format is specified, it is
+up to the driver to choose its preferred format.
+
+During the migration of a VDI, since an SR can now support different types of
+storage, we need a way to specify which format should be used. To achieve this,
+we propose adding a preferred image format for a given SR. When the SR is created,
+based on the list retrieved using `supported_image_formats` from the SM, we can
+pass `preferred-image-format=qcow2` to the `sm_config` parameter of the SR. If no
```

Comment:
This is a very roundabout way of selecting the type, and if there's migrations happening at the same time, a format for each of them doesn't seem possible.

Why can 't the VDI migration have an additional parameter selecting the type?

Reply:
In fact it is a simple way to specify a format without modifying the API but I completely agree on both point. With this modification all migrations will have the same format and it would be better to be able to choose the destination format per VDI. 
And if the user doesn't make a choice the driver that manages the destination SR makes the choice.
I will add a proposition for a new parameter for VDI pool migrate. But about VM migrate? Because the user migrate a VM that has a VDI that is not on shared storage it will call the VDI pool migrate no? As I'm not sure here what to do that's why I proposed as first approach to fix the destination format using the `preferred-image-format=qcow2`.